pkgs/build-support/appimage/appimage-exec.sh

```diff
@@ -0,0 +1,142 @@
+#!@shell@
+if [ -n "$DEBUG" ] ; then
+  set -x
+fi
+
+PATH="@path@:$PATH"
+apprun_opt=true
+
+#DEBUG=0
+
+# src : AppImage
+# dest : let's unpack() create the directory
+unpack() {
+  local src=$1
+  local out=$2
+  local appimageSignature=""
+  local appimageType=0
+
+  # https://github.com/AppImage/libappimage/blob/ca8d4b53bed5cbc0f3d0398e30806e0d3adeaaab/src/libappimage/utils/MagicBytesChecker.cpp#L45-L63
+  eval "$(r2 "$src" -nn -Nqc "p8j 3 @ 8" |
```

Comment:
Does this just extract 3 bytes at file offset 8?

Reply:
yes and it could be done with hexdump or xdd or dd. but here there is a homogeneity with r2+jq for carving elf file by design choice.